### ISSUE TITLE
`load-file-name` inconsistent when used inside macros by dependencies and emacs is started with `cask emacs` (example: geiser)

### ISSUE THREAD

**Issue by spellcard199** [cask/cask#472]
Actual problem: when a cask project depends on the `geiser` package and emacs is started with `cask emacs` it crashes when running any of the supported implementations.

`geiser` works fine when installed with `package-install`.

The following is what I think is happening.

In the `geiser` package, `define-geiser-implementation` is a macro that wraps the function `geiser-impl--define`.

The first argument to the `geiser-impl--define` function is passed by the `define-geiser-implementation` macro and should be the value of `load-file-name` at the point of its expansion.

However, when required by an emacs instance started with `cask emacs`, the value of `load-file-name` at the point where `define-geiser-implementation` is expanded becomes `"$HOME/.cask/cask-cli.el"` (where $HOME is my user home dir) instead of, for example, `"..../geiser-guile.el"` (same thing happens with other scheme implementations).

In the case of geiser the result is that
1. instead of running the correct file emacs starts running some cask code
     1. tries to contact melpa
     2. after some seconds it raises `Debugger entered--Lisp error: (bad-signature "archive-contents.sig")`
2. the library fails to be required 

Steps to reproduce:
1. Create new directory and cd into it
3. Create a new Cask file containing the following
```
(source gnu)
(source melpa)
(depends-on "geiser")
```
4. Install dependency with `cask install`
5. Run emacs with `cask emacs`
6. Require any of the supported scheme implementations:
    - `(require 'geiser-chez)`
    - `(require 'geiser-chibi)`
    - `(require 'geiser-chicken)`
    - `(require 'geiser-gambit)`
    - `(require 'geiser-guile )`
    - `(require 'geiser-mit)`
    - `(require 'geiser-racket)`

**Comment by conao3** [comment]
I cannot find your issue. Please retry on your environment?
![Screenshot_2021-02-14_06-29-59](https://user-images.githubusercontent.com/4703128/107862144-0304f500-6e8e-11eb-8867-4c2d1dbd7663.png)
![Screenshot_2021-02-14_06-29-28](https://user-images.githubusercontent.com/4703128/107862146-05674f00-6e8e-11eb-8482-1947b12fc163.png)

**Comment by spellcard199** [comment]
Hello. Sorry for late response.

If I understood/remember correctly, the "issue" was that:
1. Cask triggers compilation and therefore macro expansion during installation
2. `load-file-name`'s value when expanding macros during installation is different than what one would expect when loading a library directly from Emacs

> I cannot find your issue. Please retry on your environment?

Eventually a [micro-patch](https://gitlab.com/jaor/geiser/-/merge_requests/284/diffs?commit_id=83ad87525419ffe295b0856ff56d027eb31ad994) was merged in Geiser so it now works with Cask. It just removes the unquoting from the `load-file-name` inside the macro, avoiding the issue altogether.

After the problem with Geiser got fixed I forgot to post an update here and ask whether the `load-file-name` behavior was expected and I had to close the issue. Apologies for that.

P.S. Thanks for your work on Cask.
